src/pages/resultsView/querySummary/QuerySummary.tsx

```diff
@@ -26,15 +27,18 @@ import {buildCBioPortalPageUrl} from "../../../shared/api/urls";
 export default class QuerySummary extends React.Component<{ routingStore:ExtendedRouterStore, store: ResultsViewPageStore, onToggleQueryFormVisiblity:(visible:boolean)=>void }, {}> {
 
     @autobind
+    @action
     private toggleQueryFormVisibility() {
-        this.props.onToggleQueryFormVisiblity(this._queryFormVisible);
-        this._queryFormVisible = !this._queryFormVisible;
+        this.props.onToggleQueryFormVisiblity(this.props.store.queryFormVisible);
+        // if clicked the query button in the download tab and want to close the query form, clear the selected sample ids
+        if (this.props.store.selectedSampleIdsForQueryButton && this.props.store.queryFormVisible === true) {
```

Comment:
i think what bothers me about this is that this data belongs to a very specific part of the results view: the modify query view.  and really only to once instance of it.   How about we keep the data not on the resultsPageStore but on this view itself (QuerySummary).   Furthermore we can nest it inside of a modifyQueryParams object to make sure it's super clear what it is.   Then, you can just clear the whole thing by setting that property to undefined.  Make sense?

Reply:
Yes!

Reply:
ok, i understand your confusion.  you are right that the data needs to live at a global level (resultsPageStore).  So lets make a new property of resultsPageStore called modifyQueryParams and nest all necessary params there.   Then we just need to figure out a way to delete that object.   Lets discuss by voice.